### PR TITLE
JP-230 Error propagation documentation

### PR DESCRIPTION
**Pull request by dmggh**
Fixes https://jira.stsci.edu/browse/JP-230

### CODE REVIEW

**Comment by dmggh** [comment]
Incorporates revisions. This supercedes  PR#3772.

File: docs/jwst/error_propagation/main.rst [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+cumulative error (ERR) array values are always recomputed as the quadratic sum
+of all variances whenever a new type of variance array is added to the pipeline
+flow or whenever any of the variance arrays is updated by a step.
+
+
+Stage 1 pipelines 
+-----------------
+Stage 1 pipelines perform detector-level corrections and ramp fitting for
+individual exposures, for nearly all imaging and spectroscopic modes. Details 
+of the pipelines can be found at :ref:`Stage 1 Pipelines <calwebb_detector1>`
```
Comment (hbushouse) [inline docs/jwst/error_propagation/main.rst]: Missing a final period at the very end of the line

File: docs/jwst/error_propagation/main.rst [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+next pipeline step.
+
+Flat_field
+~~~~~~~~~~
+The SCI array from the flat-field reference file is divided into the VAR_POISSON
+and VAR_RNOISE arrays of the science data set. The variance in the flat is
+calculated in this step; the ERR array is then recalculated as these three 
+variances added in quadrature.  The newly created VAR_FLAT array and the updated 
+VAR_POISSON and VAR_RNOISE are output to the science products(s). For details of
+the flat fielding correction, see :ref:`flat_field <flat_field_step>`.
```
Comment (hbushouse) [inline docs/jwst/error_propagation/main.rst]: Need to change <flat_field_step> to <flatfield_step> (remove the first underscore). I gave you the wrong syntax in the first review 👎

File: docs/jwst/error_propagation/main.rst [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+to the next pipeline step.
+
+Flat_field
+~~~~~~~~~~
+The SCI array from the flat-field reference file is divided into the VAR_POISSON
+and VAR_RNOISE arrays of the science data set. The variance in the flat is
+calculated in this step; the ERR array is then recalculated as these three
+variances added in quadrature.  The newly created VAR_FLAT array and the updated
+VAR_POISSON and VAR_RNOISE are output to the science products(s). For details of
+the flat fielding correction, see :ref:`flat_field <flat_field_step>`.
```
Comment (hbushouse) [inline docs/jwst/error_propagation/main.rst]: Same change needed here: <flatfield_step>

**Comment by hbushouse** [comment]
Oh, in addition to adding this page into the top-level index, please also move the Data Products page up to the top level (as opposed to having it down under Package Index).

**Pull request merged**